src/main/java/app/dbService/PostgresBookService.java

```diff
@@ -120,13 +130,15 @@ public Book getBook(int id) throws DBException {
             Connection connection = getPostgresConnection();
             if (connection != null) {
                 connection.setAutoCommit(false);
+            } else {
+                throw new NullPointerException();
             }
             BookDao dao = new BookDao(connection);
             book = dao.getBook(id);
             connection.commit();
         } catch (SQLException e) {
             Logger.getGlobal().info("Connection for getting book failed.");
         }
-        return book;
+        return Optional.ofNullable(book).orElse(new Book());
```

Comment:
What I meant previously by returning Optional is returning Optional<Book> from your method. This way, you will be able to check, if optional is empty, where you call this method. Right now, by calling this method you cannot difirentiate between book being not found or not